frankenstein/ingester_lifecycle.go

```diff
@@ -30,23 +32,59 @@ const (
 	infName = "eth0"
 )
 
-// WriteIngesterConfigToConsul writes ingester config to Consul
-func WriteIngesterConfigToConsul(consulClient ConsulClient, listenPort int, numTokens int) error {
-	log.Info("Adding ingester to consul")
+// IngesterRegistration is the registration of an ingester with a register.
+type IngesterRegistration interface {
+	Unregister() error
+}
```

Comment:
That's an interesting piece. That section argues well for why it's a good idea to accept interfaces (depending on abstractions, not concretions; depending on the smallest possible thing), but "return structs" is only asserted. 

I'm not comprehensively saying returning structs is a bad idea, but I'd like to understand why it's a good idea. 

I've rolled back to returning a struct, because I'm dubious about this abstraction and am not convinced its worth dignifying with an interface. Maybe later.